chrome/content/zotero/tabs.js

```diff
@@ -85,17 +85,51 @@ var Zotero_Tabs = new function () {
 		);
 	};
 
+	this.getState = function () {
+		return this._tabs.map((tab) => {
+			var o = {
+				type: tab.type,
+				title: tab.title,
+			};
+			if (tab.data) {
+				o.data = tab.data;
+			}
+			if (tab.id == this._selectedID) {
+				o.selected = true;
+			}
+			return o;
+		});
+	};
+	
+	this.restoreState = function(tabs) {
+		for (let tab of tabs) {
+			if (tab.type === 'library') {
+				this.rename('zotero-pane', tab.title);
+			}
+			else if (tab.type === 'reader') {
+				Zotero.Reader.open(tab.data.itemID,
```

Comment:
`Zotero.Reader.open` itself now checks that.

Reply:
That doesn't seem right to me. If you call `Zotero.Reader.open()` with an invalid itemID, that should be an error. It's for the code doing the calling to know that it might not have a valid itemID and skip the open() call if so.

Reply:
This is testable synchronously with `Zotero.Items.exists(itemID)`.